Store note in secure temp while editing

leonklingele opened the pull request:
Instead temporarily store the note inside RAM.

leonklingele commented:
@bndw this also slightly changes the behavior:
We no longer check the return error of `os.Remove` (which is good imo, it shouldn't fail to save the note if it can't remove the temp file)

bndw commented:
This looks cool- can you elaborate on your license choice and implications/expectations of `pick` using `securetemp`?

bndw commented:
After chatting offline it sounds like we're ok on the licensing concerns. If you resolve the conflict I'll gladly merge.

leonklingele commented:
Rebased. I have also moved the library to my own go-gettable server. This required a change to our Makefile, as it wasn't properly handling 2-part URLs (domain/name/project vs. domain/project)

bndw commented:
Sounds good (since we're committing the vendor folder, hosting from your own server is OK with me)
